plugins/solidigm/solidigm-log-page-dir.c

```diff
@@ -0,0 +1,292 @@
+{
+	for (int index = 0; index < log_desc_size; index++) {
+		if (lid == log_desc[index].lid)
+			return log_desc[index].description;
+	}
+
+	return "Unknown";
+}
+
+static const char *lid_to_desc(const int lid, const enum solidigm_uuid uuid)
+{
+	static struct log_description standard_log_descs[] = {
+		{ 0x00, "Supported Log Pages"},
+		{ 0x01, "Error Information"},
+		{ 0x02, "SMART / Health Information"},
+		{ 0x03, "Firmware Slot Information"},
+		{ 0x04, "Changed Namespace List"},
+		{ 0x05, "Commands Supported and Effects"},
+		{ 0x06, "Device Self Test"},
+		{ 0x07, "Telemetry Host-Initiated"},
+		{ 0x08, "Telemetry Controller-Initiated"},
+		{ 0x09, "Endurance Group Information"},
+		{ 0x0A, "Predictable Latency Per NVM Set"},
+		{ 0x0B, "Predictable Latency Event Aggregate"},
+		{ 0x0C, "Asymmetric Namespace Access"},
+		{ 0x0D, "Persistent Event Log"},
+		{ 0x0E, "Predictable Latency Event Aggregate"},
+		{ 0x0F, "Endurance Group Event Aggregate"},
+		{ 0x10, "Media Unit Status"},
+		{ 0x11, "Supported Capacity Configuration List"},
+		{ 0x12, "Feature Identifiers Supported and Effects"},
+		{ 0x13, "NVMe-MI Commands Supported and Effects"},
+		{ 0x14, "Command and Feature lockdown"},
+		{ 0x15, "Boot Partition"},
+		{ 0x16, "Rotational Media Information"},
+		{ 0x70, "Discovery"},
+		{ 0x80, "Reservation Notification"},
+		{ 0x81, "Sanitize Status"},
+	};
+	const int standard_log_size = sizeof(standard_log_descs) / sizeof(struct log_description);
+
+	static struct log_description ocp_log_descs[] = {
+		{ 0xC0, "OCP SMART / Health Information Extended" },
+		{ 0xC1, "OCP Error Recovery" },
+		{ 0xC2, "OCP Firmware Activation History" },
+		{ 0xC3, "OCP Latency Monitor" },
+		{ 0xC4, "OCP Device Capabilities" },
+		{ 0xC5, "OCP Unsupported Requirements" },
+	};
+	const int ocp_log_size = sizeof(ocp_log_descs) / sizeof(struct log_description);
+
+	static struct log_description solidigm_log_descs[] = {
+		{ 0xC1, "Read Commands Latency Statistics" },
+		{ 0xC2, "Write Commands Latency Statistics" },
+		{ 0xC4, "Endurance Manager Statistics" },
+		{ 0xC5, "Temperature Statistics" },
+		{ 0xCA, "SMART Attributes" },
+	};
+	const int solidigm_log_size = sizeof(solidigm_log_descs) / sizeof(struct log_description);
+
+	static struct log_description all_vu_log_descs[] = {
+		{ 0xC0, "OCP SMART / Health Information Extended" },
+		{ 0xC1, "OCP Error Recovery or Read Commands Latency Statistics" },
+		{ 0xC2, "OCP Firmware Activation History or Write Commands Latency Statistics" },
+		{ 0xC3, "OCP Latency Monitor" },
+		{ 0xC4, "OCP Device Capabilities or Endurance Manager Statistics" },
+		{ 0xC5, "OCP Unsupported Requirements or Temperature Statistics" },
+		{ 0xCA, "SMART Attributes" },
+	};
+	const int all_vu_log_size = sizeof(all_vu_log_descs) / sizeof(struct log_description);
+
+	// Standard logs are less than 0xC0
+	if (lid < 0xC0)
+		return lid_desc_from_struct(lid, standard_log_size, standard_log_descs);
```

Comment:
Use nvme_log_to_string() here instead of lid_desc_from_struct().

Reply:
No longer applicable. Everything is now contained in lid_dir struct